docs/ceph-csi-upgrade.md

```diff
@@ -0,0 +1,334 @@
+# Ceph-csi Upgrade
+
+- [Ceph-csi Upgrade](#ceph-csi-upgrade)
+  - [Upgrading from v1.2.x to v2.0.0](#upgrading-from-v12x-to-v200)
+    - [Upgrading CephFS](#upgrading-cephfs)
+      - [1. Upgrade CephFS Provisioner resources](#1-upgrade-cephfs-provisioner-resources)
+        - [1.1 Update the CephFS Provisioner RBAC](#11-update-the-cephfs-provisioner-rbac)
+        - [1.2 Update the CephFS Provisioner deployment/statefulset](#12-update-the-cephfs-provisioner-deploymentstatefulset)
+      - [2. CSI upgrade issue](#2-csi-upgrade-issue)
+      - [3. Upgrade CephFS Nodeplugin resources](#3-upgrade-cephfs-nodeplugin-resources)
+        - [3.1 Update the CephFS Nodeplugin RBAC](#31-update-the-cephfs-nodeplugin-rbac)
+        - [3.2 Update the CephFS Nodeplugin daemonset](#32-update-the-cephfs-nodeplugin-daemonset)
+        - [3.3 Manual deletion of CephFS Nodeplugin daemonset pods](#33-manual-deletion-of-cephfs-nodeplugin-daemonset-pods)
+    - [Upgrading RBD](#upgrading-rbd)
+      - [4. Upgrade RBD Provisioner resources](#4-upgrade-rbd-provisioner-resources)
+        - [4.1 Update the RBD Provisioner RBAC](#41-update-the-rbd-provisioner-rbac)
+        - [4.2 Update the RBD Provisioner deployment/statefulset](#42-update-the-rbd-provisioner-deploymentstatefulset)
+      - [5. Upgrade RBD Nodeplugin resources](#5-upgrade-rbd-nodeplugin-resources)
+        - [5.1 Update the RBD Nodeplugin RBAC](#51-update-the-rbd-nodeplugin-rbac)
+        - [5.2 Update the RBD Nodeplugin daemonset](#52-update-the-rbd-nodeplugin-daemonset)
+        - [5.3 Manual deletion of RBD Nodeplugin daemonset pods](#53-manual-deletion-of-rbd-nodeplugin-daemonset-pods)
+
+This guide will walk you through the steps to upgrade the software in a cluster
+from v1.2.x to v2.0.0
+
+## Upgrading from v1.2.x to v2.0.0
+
+**Ceph-csi releases from master are expressly unsupported.** It is strongly
+recommended that you use [official
+releases](https://github.com/ceph/ceph-csi/releases) of Ceph-csi. Unreleased
+versions from the master branch are subject to changes and incompatibilities
+that will not be supported in the official releases. Builds from the master
+branch can have functionality changed and even removed at any time without
+compatibility support and without prior notice.
+
+### Upgrading CephFS
+
+upgrading cephfs csi includes upgrade of cephfs driver and as well as
+kubernetes sidecar containers and also the permissions required for the
+kubernetes sidecar containers, lets upgrade the things one by one
+
+git checkout latest release v2.0.0 branch
+
+```bash
+[$] git clone git@github.com:ceph/ceph-csi.git
```

Comment:
(nit) http based git checkout [links](https://github.com/ceph/ceph-csi.git) would be preferred example as otherwise a git authorization is required to clone.